wimp/constants.py

```diff
@@ -102,7 +102,7 @@
     "V2": [(3, 6), (3, 7)],
     "V3": [(3, 6), (3, 7), (4, 6)],
     "V4": [(3, 6), (3, 7), (4, 5), (4, 6)],
-    "X": [(2, 5), (2, 6), (2, 7), (2, 8)],
+    "X": [(2, 5), (2, 6), (2, 7)],
```

Comment:
Doesn't the X slot have 4 hours [here](https://erp.iitkgp.ac.in/Acad/AUT_Time_Table_24_25.pdf)?

Reply:
Technically it has 4 hours but only 3 of those are given to the lab slot X. Are there any examples where the 4th hour is also used? And if yes, how is it mentioned?

Reply:
No, because just "X" refers to the Lab Slot X (at least from our data source, i.e. "Subject List with Slots" in the ERP), which is just 3 hours long (as are all lab slots). "X4" would refer to the four hour theory slot.

Reply:
If the fourth slot is used, it would be mentioned as "X44" (fourth slot of X4)

Reply:
Example of fourth hour of X4 being used

![image](https://github.com/user-attachments/assets/2f34d4a7-cb07-4d5f-8073-2964bda36ffd)

Reply:
We should handle that case as well. Any ideas on how that could be done?

Reply:
> just "X" refers to the Lab Slot X (at least from our data source, i.e. "Subject List with Slots" in the ERP), which is just 3 hours long (as are all lab slots). "X4" would refer to the four hour slot
afaik, this is the case.
If only X is mentioned, then its the Lab Slot X
If its X with numbers, then its the courses, which can also (or only) have 4th 5-6pm slot.

Reply:
> Any ideas on how that could be done?

I guess, we can "X" mark the first 3 hours, and  "X4n" should mark the corresponding 1hour slots ??

Reply:
@rohan-b-84 I think this is already handled? I havent looked at the code, but it seems to work?
eg:  (X4 slots - wednesday 2pm to 6pm)

![image](https://github.com/user-attachments/assets/12e6912f-f8e3-4575-95c4-987e2654ab7a)

Without having looked at the code, I believe Aij is handled as `Ai[j-1]`, because Ai always is a list of slots.